src/addons/model.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @fileoverview A general modelling class used by the Presenter's
+ * getModel function.
+ */
+
+import { set, Emitter } from '../microcosm'
+
+function isObservable(binding) {
+  return binding && typeof binding.subscribe === 'function'
+}
+
+function isCallable(binding) {
+  return binding && typeof binding.call === 'function'
+}
+
+function invoke(binding, repo, scope) {
+  if (isCallable(binding)) {
+    return binding.call(scope, repo.state, repo)
+  }
+
+  return binding
+}
+
+export default class Model extends Emitter {
+  /**
+   * @param {Microcosm} repo Track this Microcosm instance for updates
+   * @param {scope} scope Scope to invoke functional bindings
+   */
+  constructor(repo, scope) {
+    super()
+
+    this.repo = repo
+    this.scope = scope
+    this.bindings = {}
+    this.subscriptions = {}
+    this.value = {}
+
+    this.repo.on('change', this.compute, this)
```

Comment:
another thing i'm curious about, it looks like `compute` could be triggered from a couple different angles. you're triggering a call here, but also in the flow of `Presenter::componentWillUpdate -> _prepareModel -> updateModel -> model.bind -> this.compute`. Would this result in duplicitous `compute` calls every time the repo state changed?